sdk/core/azure_core_amqp/src/messaging.rs

```diff
@@ -188,6 +190,121 @@ impl AmqpMessageHeader {
     pub fn delivery_count(&self) -> Option<&u32> {
         self.delivery_count.as_ref()
     }
+
+    pub fn set_durable(&mut self, durable: bool) {
```

Comment:
Why do you have setters if you also have a builder? We should have one way of doing things.

Reply:
If the builder pattern doesn't make sense in this case, we can relax requirements. Even as I was reading this, I was wondering why we need the builder pattern here when these aren't client options. The only reason we were favoring client options builders is to avoid the deep nesting we get by, say, `BlobClientOptions` containing ` StorageClientOptions` containing an `azure_core::ClientOptions` containing a `typespec_client_core::ClientOptions`, which looks hideous to type out. It's idiomatic in rust for simple data structures to just make the fields public.

Reply:
Builder makes sense, but it's slightly annoying - I need to keep a builder and a header both in my Header object (and the other objects) - the builder for the "set" APIs and the header for the "get" APIs. 

Also the builder needs to have a mutable self, and that in turn means I need to have a .clone() in the builder.build() method - again, slightly annoying.

But I'm now wondering about your suggestion of having wondering about the idea of having a "cpp_apis" feature which enables the C++ setters instead. 

While Header has only 5 fields, Properties has like 12 of them so the builder pattern makes sense for it.

Reply:
Ok, upon thinking about it, I have a better way of handling this case. I'll have to play some games with the internal semantics of the C++ wrapper but I can make it work (the wrapper APIs mimic the corresponding uAMQP APIs so I'll have some weird variable names but I can make it all work.